src/pages/settings/Wallet/ExpensifyCardPage.tsx

```diff
@@ -16,157 +16,91 @@ import useNetwork from '@hooks/useNetwork';
-        /** Field-specific server side errors keyed by microtime */
-        errorFields: PropTypes.objectOf(PropTypes.objectOf(PropTypes.string)),
+    /** The details about the Expensify cards */
+    cardList: OnyxEntry<Record<string, OnyxCard>>;
 
-        /** Field-specific pending states for offline UI status */
-        pendingFields: PropTypes.objectOf(PropTypes.objectOf(PropTypes.string)),
-    }),
-    /** User's private personal details */
-    privatePersonalDetails: PropTypes.shape({
-        legalFirstName: PropTypes.string,
-        legalLastName: PropTypes.string,
-        phoneNumber: PropTypes.string,
-        /** User's home address */
-        address: PropTypes.shape({
-            street: PropTypes.string,
-            city: PropTypes.string,
-            state: PropTypes.string,
-            zip: PropTypes.string,
-            country: PropTypes.string,
-        }),
-    }),
+    /** Draft values used by the get physical card form */
+    draftValues: OnyxEntry<GetPhysicalCardForm | undefined>;
 
-    /** Navigation route context info provided by react navigation */
-    route: PropTypes.shape({
-        params: PropTypes.shape({
-            /** domain passed via route /settings/wallet/card/:domain */
-            domain: PropTypes.string,
-        }),
-    }).isRequired,
+    /** Login info */
+    loginList: OnyxEntry<LoginList>;
 };
 
-const defaultProps = {
-    cardList: null,
-    draftValues: {
-        addressLine1: '',
-        addressLine2: '',
-        city: '',
-        state: '',
-        country: '',
-        zipPostCode: '',
-        phoneNumber: '',
-        legalFirstName: '',
-        legalLastName: '',
-    },
-    loginList: {},
-    privatePersonalDetails: {
-        legalFirstName: '',
-        legalLastName: '',
-        phoneNumber: null,
-        address: {
-            street: '',
-            city: '',
-            state: '',
-            zip: '',
-            country: '',
-        },
-    },
-};
+type ExpensifyCardPageProps = ExpensifyCardPageOnyxProps & StackScreenProps<PublicScreensParamList, typeof SCREENS.TRANSITION_BETWEEN_APPS>;
 
 function ExpensifyCardPage({
     cardList,
     draftValues,
-    loginList,
-    privatePersonalDetails,
+    loginList = {},
+    privatePersonalDetails = {},
     route: {
-        params: {domain},
+        params: {domain = ''},
     },
-}) {
+}: ExpensifyCardPageProps) {
     const styles = useThemeStyles();
     const {isOffline} = useNetwork();
     const {translate} = useLocalize();
-    const domainCards = useMemo(() => cardList && CardUtils.getDomainCards(cardList)[domain], [cardList, domain]);
-    const virtualCard = useMemo(() => (domainCards && _.find(domainCards, (card) => card.isVirtual)) || {}, [domainCards]);
-    const physicalCard = useMemo(() => (domainCards && _.find(domainCards, (card) => !card.isVirtual)) || {}, [domainCards]);
+    const domainCards = useMemo(() => CardUtils.getDomainCards(cardList ?? {})[domain], [cardList, domain]);
```

Comment:
I think previous version was not bad - when cardList was not given the function didn't call - currently it calls anyway